bin/deactivate.bat

```diff
@@ -1,36 +1,36 @@
 @echo off
+setlocal
 
-for /f "delims=" %%i in ("%~dp0..\envs") do (
-    set ANACONDA_ENVS=%%~fi
-)
-
-if "%1" == "" goto skipmissingarg
+if not "%1" == "--help" goto skipusage
+    (
     echo Usage: deactivate
     echo.
     echo Deactivates previously activated Conda
     echo environment.
+    ) 1>&2
+    exit /b 1
+:skipusage
+
+if "%1" == "" goto skiptoomanyargs
+    (echo Error: too many arguments.) 1>&2
     exit /b 1
-:skipmissingarg
+:skiptoomanyargs
 
+REM Use conda itself to figure things out
+REM No conda, no dice
 
-REM Deactivate a previous activation if it is live
-if "%CONDA_DEFAULT_ENV%" == "" goto skipdeactivate
-    REM This search/replace removes the previous env from the path
-    echo Deactivating environment "%CONDA_DEFAULT_ENV%"...
+SET CONDAFOUND=
+for %%X in (conda.exe) do (set CONDAFOUND=%%~$PATH:X)
```

Comment:
Same note as above: please do not assume conda is on PATH, but rather try to find it using relative paths and knowledge of deactivate.bat path.